packages/providers/src/sdk/CryptKeeperInjectedProvider.ts

```diff
@@ -1,372 +1,108 @@
 import {
-  IApprovals,
-  ICreateIdentityRequestArgs,
-  IConnectIdentityRequestArgs,
   ConnectedIdentityMetadata,
-  IHostPermission,
-  IInjectedMessageData,
-  IInjectedProviderRequest,
   ISemaphoreFullProof,
   ISemaphoreProofRequiredArgs,
   IRLNProofRequiredArgs,
-  IRLNSNARKProof,
+  IRLNFullProof,
   IJoinGroupMemberArgs,
   IGenerateGroupMerkleProofArgs,
   IVerifiablePresentationRequest,
 } from "@cryptkeeperzk/types";
 
-import { RPCAction } from "../constants";
-import { EventEmitter, EventHandler, EventName } from "../event";
+import { RPCExternalAction } from "../constants";
+import { Handler } from "../services";
 
-/**
- * Stores promises associated with message nonces.
- */
-const promises = new Map<string, Handlers>();
-
-const EVENTS = [
-  EventName.IDENTITY_CHANGED,
-  EventName.LOGIN,
-  EventName.LOGOUT,
-  EventName.ADD_VERIFIABLE_CREDENTIAL,
-  EventName.GENERATE_VERIFIABLE_PRESENTATION,
-  EventName.REVEAL_COMMITMENT,
-  EventName.JOIN_GROUP,
-  EventName.GROUP_MERKLE_PROOF,
-  EventName.USER_REJECT,
-];
-
-interface Handlers {
-  resolve: (res?: unknown) => void;
-  reject: (reason?: unknown) => void;
-}
+import { ICryptKeeperInjectedProvider } from "./interface";
 
 /**
  * Represents the CryptKeeper provider that is injected into the application.
  * This class is responsible for handling interactions with the CryptKeeper extension.
  *
  * @class
  */
-export class CryptKeeperInjectedProvider {
+export class CryptKeeperInjectedProvider extends Handler implements ICryptKeeperInjectedProvider {
```

Comment:
Use composition instead of inheritance.
Handler should be a class property like we have for cryptkeeper background file.